org.eclipse.sprotty/src/main/java/org/eclipse/sprotty/DefaultDiagramServer.java

```diff
@@ -455,10 +455,20 @@ protected void handle(LayoutAction action) {
 		if (getServerLayoutKind(currentRoot) == ServerLayoutKind.MANUAL) {
 			ILayoutEngine layoutEngine = getLayoutEngine();
 			if (layoutEngine != null) {
-				layoutEngine.layout(currentRoot);
+				SModelCloner cloner = getSModelCloner();
+				SModelRoot newRoot = cloner.clone(currentRoot);
```

Comment:
Could you add a comment on why cloning is necessary here?

Reply:
Added